### PR TITLE
Modularize project

### PR DESCRIPTION
**Pull request by 41zu**
The biggest change in this pull request is, that the jar now declares an module.

Also upgraded dependencies + plugins, fixed javadoc warnings and removed slf4j from the compile dependencies.

### CODE REVIEW

**Comment by 41zu** [comment]
Hi @lpradel,

I hope this message finds you well. Could you please take another look at the merge request I submitted? It would be great if you could merge it at your earliest convenience.

Just to clarify, the SLF4J dependency is only used by the TestNG test dependency, and users of the library do not need SLF4J for their usage. Here is the dependency tree for reference:

[INFO] --- dependency:3.7.0:tree (default-cli) @ steam-web-api ---
[INFO] com.lukaspradel:steam-web-api:jar:1.9.0
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.17.1:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.17.1:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.17.1:compile
[INFO] +- org.testng:testng:jar:7.10.2:test
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.36:test
[INFO] |  +- com.beust:jcommander:jar:1.82:test
[INFO] |  \- org.webjars:jquery:jar:3.7.1:test
[INFO] \- org.mockito:mockito-core:jar:5.12.0:test
[INFO]    +- net.bytebuddy:byte-buddy:jar:1.14.15:test
[INFO]    +- net.bytebuddy:byte-buddy-agent:jar:1.14.15:test
[INFO]    \- org.objenesis:objenesis:jar:3.3:test

**Comment by lpradel** [comment]
Hey, I left you a small question a while ago, was waiting for your thoughts on this. Other than that it's all good by me :)

**Comment by 41zu** [comment]
@lpradel Unfortunately, I couldn't find your question and I didn't receive any notification about it either. Could you please let me know where I can find it?

File: src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module steam.web.api {
```
[the remark this answers is not in the document]
Comment (lpradel) [reply src/main/java/module-info.java:1]: @41zu ping

[the remark this answers is not in the document]
Comment (lpradel) [reply src/main/java/module-info.java:1]: oh crap my bad it was still in pending

**Comment by 41zu** [comment]
@lpradel Your suggestion is, of course, correct. I have adopted and committed it.

**Comment by 41zu** [comment]
Thank you for merging 👍

**Pull request merged**